libsolidity/ast/Types.cpp

```diff
@@ -126,9 +126,15 @@ bool fitsPrecisionBase2(bigint const& _mantissa, uint32_t _expBase2)
 }
 
 /// Checks whether _value fits into IntegerType _type.
-bool fitsIntegerType(bigint const& _value, IntegerType const& _type)
+BoolResult fitsIntegerType(bigint const& _value, IntegerType const& _type)
 {
-	return (_type.minValue() <= _value) && (_value <= _type.maxValue());
+	if (_value < 0 && !_type.isSigned())
+		return BoolResult{std::string("Cannot implicitly convert signed literal to unsigned type.")};
+
+	if (_type.minValue() > _value || _value > _type.maxValue())
+		return BoolResult{"Literal is too large to fit in " + _type.toString(false)};
```

Comment:
The other messages had a trailing dot, this one didndt.

Reply:
ah damn, It also shows up in the test like that. Will create a PR to fix it